radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/mempool/TxStatusTest.java

```diff
@@ -0,0 +1,192 @@
+	@Inject
+	private TransactionStatusService transactionStatusService;
+
+	private final boolean shouldSucceed;
+
+	public TxStatusTest(boolean shouldSucceed) {
+		this.shouldSucceed = shouldSucceed;
+	}
+
+	private Injector createInjector() {
+		return Guice.createInjector(
+			MempoolConfig.asModule(1000, 10),
+			new BetanetForksModule(),
+			new RadixEngineOnlyLatestForkModule(View.of(100)),
+			RadixEngineConfig.asModule(1, 10, 10),
+			new SingleNodeAndPeersDeterministicNetworkModule(),
+			new MockedGenesisModule(),
+			new MempoolFillerModule(),
+			new AbstractModule() {
+				@Override
+				protected void configure() {
+					bindConstant().annotatedWith(Names.named("numPeers")).to(0);
+					bindConstant().annotatedWith(DatabaseLocation.class).to(folder.getRoot().getAbsolutePath());
+				}
+
+				@ProvidesIntoSet
+				private TokenIssuance mempoolFillerIssuance(@Self ECPublicKey self) {
+					return TokenIssuance.of(self, ValidatorStake.MINIMUM_STAKE);
+				}
+			}
+		);
+	}
+
+	public REProcessedTxn waitForCommit() {
+		var mempoolAdd = runner.runNextEventsThrough(MempoolAddSuccess.class);
+		var committed = runner.runNextEventsThrough(
+			TxnsCommittedToLedger.class,
+			c -> c.getParsedTxs().stream().anyMatch(txn -> txn.getTxn().getId().equals(mempoolAdd.getTxn().getId()))
+		);
+
+		return committed.getParsedTxs().stream()
+			.filter(t -> t.getTxn().getId().equals(mempoolAdd.getTxn().getId()))
+			.findFirst()
+			.orElseThrow();
+	}
+
+
+	public MempoolAddFailure dispatchAndCheckForError(Txn txn) {
+		mempoolAddEventDispatcher.dispatch(MempoolAdd.create(txn));
+		return runner.runNextEventsThrough(MempoolAddFailure.class);
+	}
+
+	public REProcessedTxn dispatchAndWaitForCommit(Txn txn) {
+		mempoolAddEventDispatcher.dispatch(MempoolAdd.create(txn));
+		return waitForCommit();
+	}
+
+	public REProcessedTxn dispatchAndWaitForCommit(TxAction action) {
+		nodeApplicationRequestEventDispatcher.dispatch(NodeApplicationRequest.create(action));
+		return waitForCommit();
+	}
+
+	public MempoolAddFailure dispatchAndWaitForFailure(TxAction action) {
+		nodeApplicationRequestEventDispatcher.dispatch(NodeApplicationRequest.create(action));
+		return runner.runNextEventsThrough(MempoolAddFailure.class);
+	}
+
+	@Test
+	public void singletx() throws Exception {
+		createInjector().injectMembers(this);
+		runner.start();
+
+		var accountAddr = REAddr.ofPubKeyAccount(self);
+		var otherAddr = REAddr.ofPubKeyAccount(ECKeyPair.generateNew().getPublicKey());
+
+		// Not existing transaction Id
+		var notExistingTxId = AID.from(HashUtils.random256().asBytes());
+		assertEquals(TRANSACTION_NOT_FOUND, transactionStatusService.getTransactionStatus(notExistingTxId));
+
+		// Not enought balance
+		var invalidTransfer = new TransferToken(REAddr.ofNativeToken(), accountAddr, otherAddr,
+			ValidatorStake.MINIMUM_STAKE.multiply(UInt256.TWO));
+		var invalidTransferTxId = dispatchAndWaitForFailure(invalidTransfer);
```

Comment:
Here we go to nodeApplication. It does radixEngine.construct. It fails with an exception from TransferTokensConstructorV2 (not enough balance). As it didn't get to Mempool (it's further down the stack) we never have Transaction Status FAILED

Reply:
Those calls to nodeApplication go via EventDispatcher, so I don't see how to act this exception (caught in NodeApplication https://github.com/radixdlt/radixdlt/blob/rc/1.0-beta.36/radixdlt-core/radixdlt/src/main/java/com/radixdlt/application/NodeApplication.java#L67 )